File: src/main/java/com/hohltier/decompiler/jvm/Disassembler.java
package com.hohltier.decompiler.jvm;

import org.objectweb.asm.ClassReader;
import org.objectweb.asm.ClassVisitor;
import org.objectweb.asm.util.TraceClassVisitor;
import java.io.PrintWriter;
import java.io.StringWriter;
import java.io.Writer;

public class Disassembler implements IDestructure {

    private final String code;

    public Disassembler(byte[] bytes) {
        ClassReader reader = new ClassReader(bytes);
        Writer writer = new StringWriter();
        ClassVisitor visitor = new TraceClassVisitor(new PrintWriter(writer));
        reader.accept(visitor, ClassReader.SKIP_DEBUG);
        code = writer.toString().trim();
    }

    @Override
    public String getCode() {
        return code;
    }

}

File: src/main/kotlin/dev/shota/decompiler/java/Disassembler.kt
package dev.shota.decompiler.java

import org.objectweb.asm.ClassReader
import org.objectweb.asm.util.TraceClassVisitor
import java.io.ByteArrayInputStream
import java.io.PrintWriter
import java.io.StringWriter

class Disassembler(data: ByteArray) {

    val code: String

    init {
        val reader = ClassReader(ByteArrayInputStream(data))
        val writer = StringWriter()
        val visitor = TraceClassVisitor(PrintWriter(writer))
        reader.accept(visitor, ClassReader.SKIP_DEBUG)
        code = writer.toString().trimEnd()
    }

}